geopandas/array.py

```diff
@@ -289,6 +289,34 @@ def sindex(self):
             self._sindex = get_sindex_class()(self.data)
         return self._sindex
 
+    def sindex_generated(self):
```

Comment:
How about `has_sindex`?

Reply:
I felt that that might be confusing because we already have `sindex.has_sindex`

Reply:
Am I correct in seeing that `sindex.has_sindex` was introduced in #1343?  If so, perhaps we should revisit that name as well (could be separate PR), and since it is likely internal use only, perhaps we can do so without deprecation period?

We're trying to distinguish 2 cases here:
- does the GeoDataFrame / GeoSeries / GeometryArray _have_ a spatial index that has been generated (`has_sindex` seems to indicate this most clearly)?
- does a spatial index implementation exist (maybe `has_sindex_class` or `has_sindex_backend`)

I also wonder if this should be a property rather than a function.

Reply:
I think that @brendan-ward is right here. `has_sindex` is a better name for this as a part of public facing API. While `sindex.has_index` actually check if it can have index, not if it does. 

Personally, I'd go with `has_sindex` here and `has_backend` in `sindex`. Then it makes sense reading it as:

`GeoSeries.has_index` and `GeoSeries.sindex.has_backend` (no need to repeat `sindex` in that again).

Reply:
> I also wonder if this should be a property rather than a function.

It is cheap, could be property.